lib/rewrite_rate_limit_headers.rb

```diff
@@ -9,16 +9,17 @@ def initialize(app)
   def call(request_env)
     @app.call(request_env).on_complete do |response_env|
       headers = response_env[:response_headers]
-      if headers.include? 'X-RateLimit-Limit'
+      if headers.include?('X-RateLimit-Limit') && !headers.include?('RateLimit-Limit')
         headers['RateLimit-Limit'] = headers['X-RateLimit-Limit']
       end
-      if headers.include? 'X-RateLimit-Remaining'
+      if headers.include?('X-RateLimit-Remaining') && !headers.include?('RateLimit-Remaining')
         headers['RateLimit-Remaining'] = headers['X-RateLimit-Remaining']
       end
-      if headers.include? 'X-RateLimit-Reset'
+      if headers.include?('X-RateLimit-Reset') && !headers.include?('RateLimit-Reset')
         # X-RateLimit-Reset is a timestamp (i.e. the number of seconds since epoch)
         # RateLimit-Reset is expected to be in RFC2822 format
-        headers['RateLimit-Reset'] = Time.at(headers['X-RateLimit-Reset'].to_i).rfc2822
+        # (We add 5 extra seconds for safety, in case the client & server clocks are out of sync)
```

Comment:
Can this be fixed up to the first commit? I can't see why these changes are being made in the second commit, they seem incidental.

Reply:
Good point, I think I added it later on because the spec was flaky without it, but (as you mention in another comment) it'd make more sense to add both this and the spec in a later commit